tools/src/main/java/org/apache/kafka/trogdor/workload/ProduceBenchWorker.java

```diff
@@ -179,18 +182,34 @@ protected synchronized void delay(long amount) throws InterruptedException {
 
         private final PayloadIterator values;
 
+        private final TransactionGenerator txActionGenerator;
+
         private final Throttle throttle;
 
+        private Iterator<TopicPartition> partitionsIterator;
+        private Future<RecordMetadata> sendFuture;
+        private AtomicInteger transactionsCommitted;
+        private boolean toUseTransactions = false;
+
         SendRecords(HashSet<TopicPartition> activePartitions) {
             this.activePartitions = activePartitions;
+            this.partitionsIterator = activePartitions.iterator();
             this.histogram = new Histogram(5000);
+
+            this.txActionGenerator = spec.transactionGenerator();
+            this.transactionsCommitted = new AtomicInteger();
+            if (txActionGenerator.action() == TransactionAction.INIT_TRANSACTIONS)
+                toUseTransactions = true;
+
             int perPeriod = WorkerUtils.perSecToPerPeriod(spec.targetMessagesPerSec(), THROTTLE_PERIOD_MS);
             this.statusUpdaterFuture = executor.scheduleWithFixedDelay(
-                new StatusUpdater(histogram), 30, 30, TimeUnit.SECONDS);
+                new StatusUpdater(histogram, transactionsCommitted), 30, 30, TimeUnit.SECONDS);
+
             Properties props = new Properties();
             props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, spec.bootstrapServers());
-            // add common client configs to producer properties, and then user-specified producer
-            // configs
+            if (toUseTransactions)
+                props.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, "transaction-id-" + UUID.randomUUID());
```

Comment:
The transactional ID should include `produce-bench-` in it somewhere to make it clear what test created this, right?